src/middleware.ts

```diff
@@ -1 +1,66 @@
-export function middleware(): void {}
+import { EventEmitter } from 'events'
+import { Headers } from 'headers-utils'
+import { RequestHandler as ExpressMiddleware } from 'express'
+import { RequestHandler, handleRequest, parseIsomorphicRequest } from 'msw'
+
+const emitter = new EventEmitter()
+
+export function createMiddleware(
+  ...handlers: RequestHandler[]
+): ExpressMiddleware {
+  return async (req, res, next) => {
+    const serverOrigin = `${req.protocol}://${req.get('host')}`
+    const mockedRequest = parseIsomorphicRequest({
+      id: '',
+      method: req.method,
+      url: new URL(req.url, serverOrigin),
+      headers: new Headers(req.headers as HeadersInit),
+      body: req.body,
+    })
+
+    /**
+     * @note Prepend the server's origin to each relative request handler URL.
+     *
+     * Relative URL in request handlers are usually used in JSDOM where they are
+     * rebased by MSW against the current "location" that JSDOM polyfills.
+     * In Node.js, relative request URLs is no-op as there's nothing to be relative to.
+     *
+     * @fixme Mutating handlers is not okay.
+     */
+    handlers.forEach((handler) => {
+      if ('mask' in handler.info) {
+        const mask = handler.info.mask as string
+
+        handler.info.mask = mask.startsWith('/')
+          ? new URL(handler.info.mask, serverOrigin).toString()
+          : mask
+      }
```

Comment:
Ah nice! This idea never occurred to me. I see it does indeed mutate the mask, but if it works the same then perhaps it's not really a concern?

Reply:
Mutating the handlers isn't a good idea because it'll break their re-usage. Once you invoke the middleware, it will (implicitly) alter the handler's URLs to be rebased against the server. If you combine those same handlers in another context (i.e. a Jest test), relative requests won't match anymore, as all handlers will expect to match requests issued against the server's origin.

I think the idea of some sort of URL resolving context on the `handleRequest` level is one worth exploring.

Reply:
I've added the `reusing-handlers.test.ts` to illustrate that. Right now the test will fail because `createMiddleware` alters the relative request handler URLs, making them absolute against the server. This causes the `setupServer` to skip relative requests made in the same test file.

Reply:
See https://github.com/mswjs/http-middleware/pull/1/commits/b6893111bc56b83b86ea9b446c1877def92f32ac for a possible solution